### PR TITLE
More safety when accessing index name off a dataframe

### PR DESCRIPTION
**Pull request by IanMatthewHuff**
For #4082 

-   [x] Pull request represents a single change (i.e. not fixing disparate/unrelated things in a single PR).
-   [x] Title summarizes what is changing.
-   [x] Has a [news entry](https://github.com/Microsoft/vscode-jupyter/tree/main/news) file (remember to thank yourself!).
-   [ ] Appropriate comments and documentation strings in the code.
-   [ ] Has sufficient logging.
-   [ ] Has telemetry for enhancements.
-   [ ] Unit tests & system/integration tests are added/updated.
-   [ ] [Test plan](https://github.com/Microsoft/vscode-jupyter/blob/main/.github/test_plan.md) is updated as appropriate.
-   [ ] [`package-lock.json`](https://github.com/Microsoft/vscode-jupyter/blob/main/package-lock.json) has been regenerated by running `npm install` (if dependencies have changed).

### CODE REVIEW

**Comment by IanMatthewHuff** [comment]
@joyceerhl had a comment on my previous PR here:
https://github.com/microsoft/vscode-jupyter/pull/4064#discussion_r533832608

But I didn't want to add it in there as it felt like a bit of an unrelated change (if in the same area). So new PR to make the change.

We don't have a solid repro on this, but it's based on our investigation for #4065  where we see AttributeErrors coming from this line.

I tested by opening a dataframe both from the IW variable view and from the debugger variable window. And I tried dataframes both with the default index and with an index explicitly set.

image

File: pythonFiles/vscode_datascience_helpers/dataframes/vscodeDataFrame.py [hunk 1]
```diff
@@ -69,7 +69,11 @@ def _VSCODE_getDataFrameInfo(df):
         columnNames = list(df)
 
     # Compute the index column. It may have been renamed
-    indexColumn = df.index.name if df.index.name else "index"
+    try:
+        indexColumn = df.index.name if df.index.name else "index"
```
Comment (IanMatthewHuff) [inline pythonFiles/vscode_datascience_helpers/dataframes/vscodeDataFrame.py:73]: @joyceerhl Just a slight tweak from your suggested code. Default dataframes have an index.name of None, so we still need the if statement here to revert to default "index" in that case.

**Review by DavidKutu — APPROVED** [review]

**Review by rchiodo — APPROVED** [review]

**Review by joyceerhl — APPROVED** [review]

File: pythonFiles/vscode_datascience_helpers/dataframes/vscodeDataFrame.py (hunk 1 above)
Comment (DonJayamanne) [reply pythonFiles/vscode_datascience_helpers/dataframes/vscodeDataFrame.py:73]: FYI - Please do not change `df.index.name or "index"` is also possible.

**Review by DonJayamanne — APPROVED** [review]

File: pythonFiles/vscode_datascience_helpers/dataframes/vscodeDataFrame.py (hunk 1 above)
Comment (IanMatthewHuff) [reply pythonFiles/vscode_datascience_helpers/dataframes/vscodeDataFrame.py:73]: @DonJayamanne Apologies, didn't mean to ignore this. Just missed the comment when added. I can change this later.

**Pull request merged**